pyoverkiz/const.py

```diff
@@ -1,104 +1,150 @@
         name="Hitachi Hi Kumo (Asia)",
         endpoint="https://ha203-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Hitachi",
         configuration_url=None,
+        session=session,
+        username=username,
+        password=password,
     ),
-    "hi_kumo_europe": OverkizServer(
+    "hi_kumo_europe": lambda username, password, session: DefaultClient(
         name="Hitachi Hi Kumo (Europe)",
         endpoint="https://ha117-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Hitachi",
         configuration_url=None,
+        session=session,
+        username=username,
+        password=password,
     ),
-    "hi_kumo_oceania": OverkizServer(
+    "hi_kumo_oceania": lambda username, password, session: DefaultClient(
         name="Hitachi Hi Kumo (Oceania)",
         endpoint="https://ha203-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Hitachi",
         configuration_url=None,
+        session=session,
+        username=username,
+        password=password,
     ),
-    "nexity": OverkizServer(
+    "nexity": lambda username, password, session: NexityClient(
         name="Nexity Eugénie",
         endpoint="https://ha106-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Nexity",
         configuration_url=None,
+        session=session,
+        username=username,
+        password=password,
     ),
-    "rexel": OverkizServer(
+    "rexel": lambda username, password, session: DefaultClient(
         name="Rexel Energeasy Connect",
         endpoint="https://ha112-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Rexel",
         configuration_url="https://utilisateur.energeasyconnect.com/user/#/zone/equipements",
+        session=session,
+        username=username,
+        password=password,
     ),
-    "simu_livein2": OverkizServer(  # alias of https://tahomalink.com
+    "simu_livein2": lambda username, password, session: DefaultClient(  # alias of https://tahomalink.com
         name="SIMU (LiveIn2)",
         endpoint="https://ha101-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Somfy",
         configuration_url=None,
+        session=session,
+        username=username,
+        password=password,
     ),
-    "somfy_europe": OverkizServer(  # alias of https://tahomalink.com
+    "somfy_europe": lambda username, password, session: SomfyClient(  # alias of https://tahomalink.com
         name="Somfy (Europe)",
         endpoint="https://ha101-1.overkiz.com/enduser-mobile-web/enduserAPI/",
         manufacturer="Somfy",
         configuration_url="https://www.tahomalink.com",
+        session=session,
+        username=username,
+        password=password,
+    ),
+    "somfy_europe_local": lambda gateway_id, token, session: SomfyLocalClient(
+        name="Somfy Local(Europe)",
+        endpoint=f"https://gateway-{gateway_id}.local:8443/enduser-mobile-web/1/enduserAPI/",
```

Comment:
This is not flexible enough I would say. There will be users (like me) where bonjour is not working (e.g. they have a VPN / different VNET). It would be good if we can specify a full URL, thus an IP or a .local address is both possible.

Reply:
Hum, that’s good to now. So on HA side, you have to enter manually the IP of your box?

Reply:
Yes